tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -0,0 +1,75 @@
+using Mono.Cecil;
+using Mono.Cecil.Rocks;
+using System.Linq;
+
+using Microsoft.MaciOS.Nnyeah.AssemblyComparator;
+
+namespace Microsoft.MaciOS.Nnyeah {
+    // Converting constructors from Legacy to NET6 have some special concerns:
+    //     - The arguments to NSObject's base class (IntPtr) and (IntPtr, bool) must be moved to NativeHandle, but _only_ if we're derived from NSObject
```

Comment:
There's an implicit conversion from IntPtr to NativeHandle, which means that another way to modify these constructors would be to not change the signature, but only the invocation to the base type's NativeHandle ctor by injecting a call to the implicit conversion operator first.

So something like this:

```cs
public void C : NSObject {
    public C (IntPtr handle)
        : base (handle)
    {
    }
}
```

would become:

```cs
public class C {
    public C (IntPtr handle)
        : ((NativeHandle) handle)
    {
    }
```